phoenix-pherf/src/main/java/org/apache/phoenix/pherf/jmx/MonitorManager.java

```diff
@@ -51,13 +58,11 @@
     private volatile boolean shouldStop = false;
```

Comment:
shouldStop doesn't need to be volatile. You are already synchronizing accesses to it. Better would be to make shouldStop final and an AtomicBoolean.